### PR TITLE
fix: subscriptions bugs

### PR DESCRIPTION
**Pull request by helenosheaa**
Closes https://github.com/influxdata/data-acquisition/issues/226
Closes https://github.com/influxdata/data-acquisition/issues/227
Closes https://github.com/influxdata/data-acquisition/issues/256
Closes https://github.com/influxdata/data-acquisition/issues/228
Closes https://github.com/influxdata/data-acquisition/issues/257

Edit unselected:
Screen Shot 2022-04-20 at 2 47 03 PM

Edit selected:
Screen Shot 2022-04-20 at 2 47 10 PM

### CODE REVIEW

File: src/writeData/subscriptions/components/BrokerFormContent.tsx [hunk 1]
```diff
@@ -265,7 +274,7 @@ const BrokerFormContent: FC<Props> = ({updateForm, formContent, className}) => {
               }}
               value="none"
               titleText="None"
-              disabled={false}
+              disabled={edit ? false : true}
```
Comment (genehynson) [inline src/writeData/subscriptions/components/BrokerFormContent.tsx:277]:
```suggestion
              disabled={!edit}
```

File: src/writeData/subscriptions/components/BrokerFormContent.tsx [hunk 2]
```diff
@@ -284,7 +293,7 @@ const BrokerFormContent: FC<Props> = ({updateForm, formContent, className}) => {
               }}
               value="user"
               titleText="User"
-              disabled={false}
+              disabled={edit ? false : true}
```
Comment (genehynson) [inline src/writeData/subscriptions/components/BrokerFormContent.tsx:296]:
```suggestion
              disabled={!edit}
```

File: src/writeData/subscriptions/components/JsonPathInput.tsx [hunk 3]
```diff
@@ -143,8 +154,14 @@ const JsonPathInput: FC<Props> = ({name, formContent, updateForm, itemNum}) => {
...
                   testID={`${tagType}-json-parsing-type`}
+                  status={
+                    edit ? ComponentStatus.Default : ComponentStatus.Disabled
+                  }
                 >
-                  {dataType}
+                  {tagType
+                    ? sanitizeType(formContent.jsonTagKeys[itemNum].type)
+                    : sanitizeType(formContent.jsonFieldKeys[itemNum].type) ??
+                      dataType}
```
Comment (genehynson) [inline src/writeData/subscriptions/components/JsonPathInput.tsx:164]: is the `dataType` default just supposed to apply to the `sanitizeType(formContent.jsonFieldKeys[itemNum].type)` statement and not the tags one?

File: src/writeData/subscriptions/components/SubscriptionForm.tsx [hunk 4]
```diff
@@ -96,7 +96,7 @@ const SubscriptionForm: FC<Props> = ({
               currentSubscription={formContent}
               updateForm={updateForm}
               className="create"
-              edit={false}
+              edit={true}
```
Comment (genehynson) [inline src/writeData/subscriptions/components/SubscriptionForm.tsx:99]: edit is on by default?

File: src/writeData/subscriptions/utils/form.ts [hunk 5]
```diff
@@ -25,6 +25,9 @@ export const sanitizeForm = (form: Subscription): Subscription => {
     if (newVal) {
       form.jsonMeasurementKey.path = newVal
     }
+    if (form.jsonMeasurementKey.type === 'number') {
+      form.jsonMeasurementKey.type = 'double'
+    }
   }
```
Comment (genehynson) [inline src/writeData/subscriptions/utils/form.ts:31]: will this work for when we load an existing subscription into the details page? Will the dropdown still show "Number"?

File: src/writeData/subscriptions/components/SubscriptionForm.tsx (hunk 4 above)
Comment (helenosheaa) [reply src/writeData/subscriptions/components/SubscriptionForm.tsx:99]: the same components are used for create and edit, so this one is a create instance so always editable

File: src/writeData/subscriptions/utils/form.ts (hunk 5 above)
Comment (helenosheaa) [reply src/writeData/subscriptions/utils/form.ts:31]: yes i've made the logic so it returns number if its double - thats what the sanitizeType function is doing.

File: src/writeData/subscriptions/components/JsonPathInput.tsx (hunk 3 above)
Comment (helenosheaa) [reply src/writeData/subscriptions/components/JsonPathInput.tsx:164]: no, good catch

File: src/writeData/subscriptions/components/SubscriptionForm.tsx (hunk 4 above)
Comment (genehynson) [reply src/writeData/subscriptions/components/SubscriptionForm.tsx:99]: ah, got it!

File: src/writeData/subscriptions/utils/form.ts (hunk 5 above)
Comment (genehynson) [reply src/writeData/subscriptions/utils/form.ts:31]: got it!

**Review by genehynson — APPROVED** [review]

**Pull request merged**